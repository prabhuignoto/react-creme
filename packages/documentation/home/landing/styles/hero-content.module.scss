@use '@design/theme.scss';

.content {
  border-radius: 8px;
  padding: 3rem;
}

@mixin grid {
  column-gap: 2rem;
  display: grid;
  grid-template-columns: 1fr 2fr;
  justify-items: center;
}

@media (min-width: 768px) {
  .content {
    max-width: 600px;
  }
}

@media (min-width: 1024px) {
  .content {
    display: flex;
    flex-direction: column;
    max-width: 800px;
  }
}

@media (min-width: 1280px) {
  .content {
    @include grid;
    max-width: 1000px;
  }
}

@media (min-width: 1440px) {
  .content {
    @include grid;
    max-width: 1240px;
  }
}

@media (min-width: 1920px) {
  .content {
    @include grid;
    max-width: 1700px;
  }
}

@media (min-width: 2560px) {
  .content {
    @include grid;
    max-width: 2000px;
  }
}

.left_column {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  min-width: 400px;
}

.right_column {
  align-items: center;
  align-self: flex-end;
  display: grid;
  gap: 1rem;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  width: 100%;
}

.header {
  color: theme.$primary;
  font-size: 3rem;
  font-weight: 500;
  margin: 0.2rem 0;
  text-transform: uppercase;
}

.sub_text {
  font-size: 2rem;
  font-weight: 300;
  margin: 1rem 0;

  &.dark {
    color: #fff;
  }
}

.get_started_btn {
  background-color: var(--rc-secondary-color-hex);
  border: 0;
  border-radius: 4px;
  color: var(--rc-color-white);
  font-size: 1.2rem;
  margin-top: 3rem;
  padding: 1rem;
  width: 200px;
}

.buttons {
  display: flex;
  margin: 3rem 0 0;

  > :first-child {
    margin-right: 1rem;
  }
}
