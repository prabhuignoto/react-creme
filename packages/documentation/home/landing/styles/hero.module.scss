.hero {
  align-items: flex-start;
  display: flex;
  justify-content: center;
  width: 100%;

  @media (min-width: 1rem) {
    min-height: 1200px;
  }

  @media (min-width: 43rem) {
    min-height: 980px;
  }

  @media (min-width: 768px) {
    min-height: 1080px;
  }

  @media (min-width: 1280px) {
    min-height: 820px;
  }

  &:not(.dark) {
    background: linear-gradient(
      to left bottom,
      rgba(#bfdbf7, 1) 20%,
      rgba(#e1e5f2, 1)
    );
  }

  &.dark {
    background: linear-gradient(
      to right bottom,
      rgba(#b8c0ff, 0.3) 10%,
      rgba(#1e1e1e, 0.9)
    );
  }
}
