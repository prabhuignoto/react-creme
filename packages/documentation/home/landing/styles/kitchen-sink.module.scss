.sink {
  align-items: center;
  // box-shadow: 0px 0px 10px 5px rgba(#000, 0.1);
  background: #fff;
  border-radius: 4px;
  display: grid;
  height: 300px;
  justify-items: center;
  justify-items: center;
  padding: 1rem;
  width: calc(100% - 2rem);

  &.dark {
    background: #1e1e1e;
  }
}

.sink1 {
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr 1fr;
}

.sink2 {
  grid-template-columns: 1fr;
}

.sink3 {
  grid-column: 1 / 3;
  grid-template-columns: 1fr 1fr;
  height: 350px;
}

.widget {
  align-items: center;
  border-radius: 4px;
  // padding: 1rem;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 90%;
}

.widget_wrap {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  width: 100%;
}

.widget_wrap_2 {
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  width: 100%;
}
