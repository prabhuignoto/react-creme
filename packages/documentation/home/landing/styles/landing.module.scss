.landing {
  align-items: center;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.features_wrapper {
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;

  @media (min-width: 768px) {
    // padding-top: 3rem;
  }
}
