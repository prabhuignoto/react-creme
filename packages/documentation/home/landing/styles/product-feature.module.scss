@use '@design/theme.scss';

.feature {
  align-items: flex-start;
  border-radius: 6px;
  color: #fff;
  display: flex;
  flex-direction: column;
  height: 100px;
  height: 160px;
  justify-content: flex-start;
  margin: 0.5rem 0;
  padding: 1rem;
  width: calc(100% - 2rem);

  @media (min-width: 768px) {
    height: 150px;
    margin: 0;
  }

  &:not(.dark) {
    background: linear-gradient(
        to left top,
        rgba(#b8c0ff, 0.5) 5%,
        rgba(#fff, 0.85)
      ),
      linear-gradient(to right bottom, rgba(#b8c0ff, 0.5) 9%, rgba(#fff, 0.5));
  }

  &.dark {
    background: linear-gradient(
        to left top,
        rgba(#1e1e1e, 0.5) 5%,
        rgba(#4a4a4a, 0.5)
      ),
      linear-gradient(
        to right bottom,
        rgba(#1e1e1e, 0.5) 9%,
        rgba(#4a4a4a, 0.5)
      );
  }
}

.title {
  font-size: 1.2rem;
  font-weight: 500;
  margin: 1rem 0;
  padding-left: 2rem;
  text-transform: capitalize;

  @media (min-width: 768px) {
    font-size: 1.5rem;
    font-weight: 600;
  }

  &:not(.dark) {
    color: var(--rc-primary-color-hex);
  }

  &.dark {
    color: var(--rc-primary-color-hex);
  }
}

.icon {
  align-items: center;
  color: #fff;
  display: flex;
  height: 1.25rem;
  justify-content: center;
  position: relative;
  width: 1.25rem;
  z-index: 1;

  @media (min-width: 768px) {
    height: 1.5rem;
    width: 1.5rem;
  }

  &:not(.dark)::after {
    background: linear-gradient(
      to top right,
      var(--rc-primary-color-hex),
      var(--rc-secondary-color-hex)
    );
  }

  &.dark::after {
    background: var(--rc-primary-color-hex);
  }

  &::after {
    border-radius: 50%;
    content: '';
    display: block;
    // background-color: var(--rc-primary-color-hex);
    height: 2.5rem;
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translateX(-50%) translateY(-50%);
    width: 2.5rem;
    z-index: -1;

    @media (min-width: 768px) {
      height: 3rem;
      width: 3rem;
    }
  }

  svg {
    height: 100%;
    width: 100%;
  }
}

.description {
  align-items: center;
  color: var(--rc-primary-color-hex);
  display: flex;
  font-size: 1.25rem;
  font-weight: 300;
  justify-content: center;
  margin-top: 0.5rem;
  text-align: left;

  @media (min-width: 768px) {
    font-size: 1.25rem;
  }

  &:not(.dark) {
    color: var(--rc-primary-color-hex);
  }

  &.dark {
    color: theme.$steel-gray;
  }
}

.features {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 1rem;
  width: 100%;

  @media (min-width: 768px) {
    column-gap: 1rem;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr 1fr;
    justify-items: center;
    list-style: none;
    row-gap: 2rem;
  }
}

.feature_item {
  width: calc(100% - 4rem);
}

@media (min-width: 768px) {
  .features {
    max-width: 100%;
  }
}

@media (min-width: 1024px) {
  .features {
    max-width: 950px;
  }
}

@media (min-width: 1280px) {
  .features {
    max-width: 1024px;
  }
}

@media (min-width: 1536px) {
  .features {
    max-width: 1300px;
  }
}

.header {
  align-items: center;
  display: flex;
  justify-content: center;
  padding-left: 1rem;
}
