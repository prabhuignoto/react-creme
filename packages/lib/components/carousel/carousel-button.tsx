import classNames from 'classnames';
import React from 'react';
import { ChevronRightIcon } from '../../icons';
import { Button } from '../button/button';
import { CarouselButtonProps } from './carousel-model';
import './carousel.scss';

const CarouselButton: React.FunctionComponent<CarouselButtonProps> = ({
  onClick,
  position,
  hide,
  direction,
  focusable,
  label,
}: CarouselButtonProps) => {
  return (
    <span
      className={classNames(
        [
          'rc-carousel-btn',
          `rc-carousel-btn-${position}`,
          `rc-carousel-btn-${direction}`,
        ],
        {
          'rc-carousel-btn-hide': hide,
        }
      )}
    >
      <Button type="icon" onClick={onClick} focusable={focusable} label={label}>
        <ChevronRightIcon />
      </Button>
    </span>
  );
};

CarouselButton.displayName = 'CarouselButton';

export { CarouselButton };
