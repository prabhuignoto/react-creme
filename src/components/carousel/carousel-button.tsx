import classNames from "classnames";
import React from "react";
import { Button } from "..";
import { ChevronRightIcon } from "../../icons";
import "./carousel.scss";

interface CarouselButtonModel {
  onClick: () => void;
  position: "left" | "right";
}

const CarouselButton: React.FunctionComponent<CarouselButtonModel> = React.memo(
  ({ onClick, position }: CarouselButtonModel) => {
    return (
      <span
        className={classNames([
          "rc-carousel-btn",
          `rc-carousel-btn-${position}`,
        ])}
      >
        <Button type="icon" onClick={onClick}>
          <ChevronRightIcon />
        </Button>
      </span>
    );
  }
);

CarouselButton.displayName = "CarouselButton";

export { CarouselButton };
