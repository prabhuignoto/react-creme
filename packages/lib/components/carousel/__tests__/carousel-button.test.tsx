import { fireEvent, render } from '@testing-library/react';
import { vi } from 'vitest';
import { CarouselButton } from '../carousel-button';
import styles from '../carousel.module.scss';

const handler = vi.fn();

describe('Carousel Button', () => {
  it('should render with position', async () => {
    const { container } = render(
      <CarouselButton
        position="left"
        onClick={handler}
        direction="horizontal"
        label="Previous"
      />
    );

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveClass(styles.btn_left);
  });

  it('should render carousel button snapshot', async () => {
    const { getByRole } = render(
      <CarouselButton
        position="left"
        onClick={handler}
        direction="vertical"
        label="previous"
      />
    );

    expect(getByRole('button')).toMatchSnapshot();
  });

  it('should call the onClick handler', async () => {
    const { getByRole } = render(
      <CarouselButton
        position="left"
        onClick={handler}
        direction="horizontal"
        label="Previous"
      />
    );

    fireEvent.click(getByRole('button'));

    expect(handler).toBeCalled();
  });
});
