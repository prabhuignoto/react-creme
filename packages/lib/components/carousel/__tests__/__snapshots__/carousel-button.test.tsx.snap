// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Button should render carousel button snapshot 1`] = `
<button
  aria-label="previous"
  class="rc-btn-sm rc-btn-icon rc-btn"
  role="button"
  style="outline: none;"
  tabindex="0"
>
  <span
    class="rc-btn-icon-container"
  >
    <svg
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 18l6-6-6-6"
      />
    </svg>
  </span>
</button>
`;
